src/mudlet-lua/lua/GUIUtils.lua

```diff
@@ -597,7 +597,7 @@ end
 function replaceAll(word, what)
   local startp, endp = 1, 1
   while true do
-    startp, endp = getCurrentLine():find(word, endp + (#what - #word) + 1)
+    startp, endp = getCurrentLine():find(word)
```

Comment:
Does this removal impact on the _recursive_ replacement functionality?  I'm not an expert on Lua string handling, but I am suspicious that that extra argument to `string:find()` is needed in some conditions, if the lengths of the `word` and `what` are constant (which they seem to be) then they could be replaced by a single variable calculated outside of the loop. My guess to the issue found is that repeated replacement of a long search term by a short one or vis-versa is hammering the memory allocation process as each instance of `word` is replaced by `what`...

Reply:
It does break recursive functionality. 
The issue, before the patch, seems to be that `endp + (#what - #word) + 1` will result in a negative value if `#word` is sufficiently larger than `#what`, or if the result of `endp + (#what - #word) + 1` indexes a location after an occurrence of `word`. This latter case would occur if `what` was sufficiently larger than `word`. This seems to only be an issue on first checking for `word`; subsequent checks (if they were to occur) would behave as expected.